network/ServerNetworking.cpp

```diff
@@ -112,6 +112,10 @@ bool PlayerConnection::SendMessage(const Message& message) {
     return SyncWriteMessage(message);
 }
 
+bool PlayerConnection::IsEstablished() const {
+    return (m_ID != INVALID_PLAYER_ID || !m_player_name.empty() || m_client_type != Networking::INVALID_CLIENT_TYPE);
+}
+
```

Comment:
Should be client type tested here? I'm going to set it before establishing between authentication request and response.